scripts/installer.py

```diff
@@ -0,0 +1,161 @@
+                logging.debug(f'Command output: {stdout.decode(locale.getpreferredencoding(False))}')
+
+            if process.returncode != 0:
+                self.printErrorAndExit(f"Command failed with error : {stderr.decode(locale.getpreferredencoding(False))}")
+
+            return stdout.decode(locale.getpreferredencoding(False))
+        except OSError as err:
+            self.printErrorAndExit(f"Error while executing command: {err}")
+
+    """
+    Downloads the file at the given url and saves it to the given filename
+    """
+    def downloadFile(self, url, filename):
+        logging.debug(f'Downloading {filename} from {url}')
+        try:
+            sslContext = None
+            if platform.system() != "Windows":
+                import certifi
+                sslContext = ssl.create_default_context(cafile=certifi.where()) 
+            with urllib.request.urlopen(url, context=sslContext) as response, open(filename, 'wb') as out_file:
+                shutil.copyfileobj(response, out_file)
+        except URLError as err:
+            logging.info("Failed to download " + filename + " :")
+            self.printErrorAndExit(err)
+        logging.info(filename + " downloaded successfully")
+
+    """
+    This method must be implemented by the installer of each platform to install and configure the DNS server.
+    """
+    def setupDNS(self):
+        pass
+
+    """
+    Generates an HTTPS certificate for my.massa using mkcert and stores it in the thyra config folder.
+    """
+    def generateCertificate(self):
+        logging.info("Generating HTTPS certificate for my.massa")
+
+        if not os.path.exists(self.CERTIFICATIONS_FOLDER_PATH):
+            try:
+                os.makedirs(self.CERTIFICATIONS_FOLDER_PATH)
+            except OSError as err:
+                self.printErrorAndExit(f"Error while creating certs folder: {err}")
+
+        self.downloadFile(self.MKCERT_URL, self.MKCERT_FILENAME)
+        os.chmod(self.MKCERT_FILENAME, 0o755)
+
+        self.executeCommand([
+            os.path.join(os.getcwd(), self.MKCERT_FILENAME), 
+            "--install"])
+        self.executeCommand([
+            os.path.join(os.getcwd() , self.MKCERT_FILENAME),
+            "--cert-file", os.path.join(self.CERTIFICATIONS_FOLDER_PATH , "cert.pem"),
+            "--key-file", os.path.join(self.CERTIFICATIONS_FOLDER_PATH, "cert-key.pem"),
+            "my.massa"])
+
+        try:
+            os.remove(self.MKCERT_FILENAME)
+        except OSError as err:
+            self.printErrorAndExit(f"Error while deleting mkcert binary: {err}")
+        logging.info("HTTPS certificate successfully generated")
+
+    """
+    Downloads thyra server and stores it in the thyra install folder.
+    """
+    def installThyraServer(self):
+        self.downloadFile(self.THYRA_SERVER_URL, self.THYRA_SERVER_FILENAME)
+        os.chmod(self.THYRA_SERVER_FILENAME, 0o755)
+        if os.getcwd() != self.THYRA_INSTALL_FOLDER_PATH:
```

Comment:
And if not you install nothing  without any log ?
Why are you doing this test ?

Reply:
If not, we don't need to do what's inside of the if since we already installed Thyra in the right place.

Reply:
Thyra is installed, but is it the last version ?

Reply:
I can add a check for that by running thyra using the `--version`.

Reply:
You can. You can also systematically reinstall thyra if it's done faster as this is not explicitly ask for this task.

@SlnPons wdyt?

Reply:
@gregLibert @thomas-senechal the best action for the user will always be the fastest when it comes to installing sth. As long as it is cleaned which means that we don't install twice or take more space that we should. 
Does it help @thomas-senechal ?

Reply:
I think we should keep it as it is, so, reinstalling Thyra at each run. Users might restart the installer if something failed or to update Thyra. 

If we add a check, we should either:
- add an option to force reinstall => isn't a good idea since the purpose of the installer is to be used without a terminal.  
- ask the user if he wants to perform a reinstallation or Thyra, and then, of Thyra App (icon tray). But that would also require us to check for the latest version of Thyra locally and on Github.

I think for now, we should keep it as it is and reinstallation every time the script is run. 

WDYT @gregLibert @SlnPons ?